test/Numerics/DGMethods/bickley_jet/TwoDimensionalCompressibleNavierStokesEquations.jl

```diff
@@ -0,0 +1,498 @@
+    t::Real,
+    direction,
+)
+    coriolis_force!(model, model.coriolis, source, state, aux, t)
+    forcing_term!(model, model.forcing, source, state, aux, t)
+    linear_drag!(model, model.turbulence, source, state, aux, t)
+
+    return nothing
+end
+
+coriolis_force!(::CNSE2D, ::Nothing, _...) = nothing
+
+@inline function coriolis_force!(
+    model::CNSE2D,
+    coriolis::fPlaneCoriolis,
+    source,
+    state,
+    aux,
+    t,
+)
+    ρu = @SVector [state.ρu[1], state.ρu[2], -0]
+
+    # f × u
+    f = [-0, -0, coriolis_parameter(model, coriolis, aux.coords)]
+    id = @SVector [1, 2]
+    fxρu = (f × ρu)[id]
+
+    source.ρu -= fxρu
+
+    return nothing
+end
+
+forcing_term!(::CNSE2D, ::Nothing, _...) = nothing
+
+@inline function forcing_term!(
+    model::CNSE2D,
+    forcing::KinematicStress,
+    source,
+    state,
+    aux,
+    t,
+)
+    source.ρu += kinematic_stress(model, forcing, aux.coords)
+
+    return nothing
+end
+
+linear_drag!(::CNSE2D, ::ConstantViscosity, _...) = nothing
+
+@inline function linear_drag!(::CNSE2D, turb::LinearDrag, source, state, aux, t)
+    source.ρu -= turb.λ * state.ρu
+
+    return nothing
+end
+
+@inline wavespeed(m::CNSE2D, _...) = m.c
+
+roe_average(ρ⁻, ρ⁺, var⁻, var⁺) =
+    (sqrt(ρ⁻) * var⁻ + sqrt(ρ⁺) * var⁺) / (sqrt(ρ⁻) + sqrt(ρ⁺))
+
+function numerical_flux_first_order!(
+    ::RoeNumericalFlux,
+    model::CNSE2D,
+    fluxᵀn::Vars{S},
+    n⁻::SVector,
+    state⁻::Vars{S},
+    aux⁻::Vars{A},
+    state⁺::Vars{S},
+    aux⁺::Vars{A},
+    t,
+    direction,
+) where {S, A}
+    numerical_flux_first_order!(
+        CentralNumericalFluxFirstOrder(),
+        model,
+        fluxᵀn,
+        n⁻,
+        state⁻,
+        aux⁻,
+        state⁺,
+        aux⁺,
+        t,
+        direction,
+    )
+
+    FT = eltype(fluxᵀn)
+
+    # constants and normal vectors
+    g = model.g
+    @inbounds nˣ = n⁻[1]
+    @inbounds nʸ = n⁻[2]
+
+    # get minus side states
+    ρ⁻ = state⁻.ρ
+    @inbounds ρu⁻ = state⁻.ρu[1]
+    @inbounds ρv⁻ = state⁻.ρu[2]
+    ρθ⁻ = state⁻.ρθ
+
+    # get plus side states
+    ρ⁺ = state⁺.ρ
+    @inbounds ρu⁺ = state⁺.ρu[1]
+    @inbounds ρv⁺ = state⁺.ρu[2]
+    ρθ⁺ = state⁺.ρθ
+
+    # averages for roe fluxes
+    ρ = (ρ⁺ + ρ⁻) / 2
+    ρu = (ρu⁺ + ρu⁻) / 2
+    ρv = (ρv⁺ + ρv⁻) / 2
+    ρθ = (ρθ⁺ + ρθ⁻) / 2
+
+    # derived averages
+    u = ρu / ρ
+    v = ρv / ρ
+    θ = ρθ / ρ
+    uₙ = nˣ * u + nʸ * v
+
+    # differences for difference vector
+    Δρ = ρ⁺ - ρ⁻
+    Δρu = ρu⁺ - ρu⁻
+    Δρv = ρv⁺ - ρv⁻
+    Δρθ = ρθ⁺ - ρθ⁻
+
+    Δφ = [Δρ Δρu Δρv Δρθ]
+
+    # jacobian
+    ∂F∂φ = [
+        0 nˣ nʸ 0
+        (nˣ * g * ρ - u * uₙ) (uₙ + nˣ * u) (nʸ * u) 0
+        (nʸ * g * ρ - v * uₙ) (nˣ * v) (uₙ + nʸ * v) 0
+        (-θ * uₙ) (nˣ * θ) (nʸ * θ) uₙ
+    ]
+
+    # actually calculate flux
+    flux = ∂F∂φ * Δφ'
+
+    @inbounds begin
+        fluxᵀn.ρ -= flux[1]
+        fluxᵀn.ρu -= @SVector [flux[2], flux[3]]
+        fluxᵀn.ρθ -= flux[4]
+    end
```

Comment:
@jkozdon @simonbyrne is there a smarter way to do this matrix multiply?

Reply:
Probably make it an `SMatrix` and an `SVector`

Reply:
that works and @charleskawczynski showed me how to use parent to get it in one line